### PR TITLE
Feature/6 add simple auth

### LINKED ISSUE #6: Add auth
## Problem

We want to have authentication and authorization (if needed) in the application. Right now we use Exasol user credentials for each query.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #6

### CODE REVIEW

**Review by pj-spoelders — APPROVED** [review]
Looks good

File: pkg/exasol-rest-api/authorizer.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	authorized := false
+	for _, token := range tokens {
+		if auth.AllowedTokens[token] {
+			authorized = true
+			break
+		}
+	}
+
+	if !authorized {
+		return fmt.Errorf(error_reporting_go.ExaError("E-ERA-22").
```
Comment (jakobbraun) [inline pkg/exasol-rest-api/authorizer.go:32]: Wait 100ms here to slow down brute force attacks.

File: pkg/exasol-rest-api/authorizer.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			Message("an authorization token is missing or wrong.").
+			Mitigation("please make sure you provided a valid token.").Error())
```
Comment (jakobbraun) [inline pkg/exasol-rest-api/authorizer.go:34]: Maybe log failed login attemps

File: doc/user_guide/user-guide.md [hunk 3]
```diff
@@ -41,6 +42,19 @@ Before starting the application, you need to set an environment variable that po
...
 
+### Authorization
+
+Add a list of API tokens to the configuration file:
+
+```yaml
+api-tokens:
+  - "Z9wsPLOpKV"
+  - "olwU1t7w4F"
+  - "Br3Bb822tv"
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply doc/user_guide/user-guide.md:53]: Please also add a verification for a minimum token length of 30 characters. It's too easy too brute force...

**Review by jakobbraun — CHANGES_REQUESTED** [review]

**Comment by jakobbraun** [comment]
As far as I understood, customers will run  this tool on the internet, so that power apps can access it?
In that case I think we should make SSL mandatory or strongly advocate it (for example name the property to disable it `insecure-disable-tls`).

**Comment by jakobbraun** [comment]
@kaklakariada can you also take a look since it's security critical

**Comment by AnastasiiaSergienko** [comment]
> As far as I understood, customers will run this tool on the internet, so that power apps can access it? In that case I think we should make SSL mandatory or strongly advocate it (for example name the property to disable it `insecure-disable-tls`).

Let's discuss it in this ticket. Feel free to add your concerns and ideas: https://github.com/exasol/exasol-rest-api/issues/22

File: pkg/exasol-rest-api/authorizer.go (hunk 2 above)
Comment (AnastasiiaSergienko) [reply pkg/exasol-rest-api/authorizer.go:34]: added log message

File: doc/user_guide/user-guide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply doc/user_guide/user-guide.md:53]: done

File: pkg/exasol-rest-api/authorizer.go (hunk 1 above)
Comment (AnastasiiaSergienko) [reply pkg/exasol-rest-api/authorizer.go:32]: The API deserves to have rate limits/throttling -> added to https://github.com/exasol/exasol-rest-api/issues/22

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**